src/v3/plugins/github/relationalView.test.js

```diff
@@ -1,70 +1,162 @@
 // @flow
 
 import * as R from "./relationalView";
+import * as N from "./nodes";
 
 describe("plugins/github/relationalView", () => {
   const data = require("./demoData/example-github");
   // Sharing this state is OK because it's just a view - no mutation allowed!
   const view = new R.RelationalView(data);
 
-  function assertNotNull<T>(x: ?T): T {
-    if (x == null) {
-      throw new Error(`Assertion fail: ${String(x)} `);
-    }
-    return x;
+  function hasEntities(name, method) {
+    describe(name, () => {
+      const all = Array.from(method());
+      it(`has expected number of ${name}`, () => {
+        expect(all.length).toMatchSnapshot();
+      });
+      const one = all[0];
+      it("have expected type", () => {
+        expect(one.type()).toMatchSnapshot();
+      });
+      it("all have same type", () => {
+        const types = all.map((x) => x.type());
+        expect(new Set(types).size).toBe(1);
+      });
+      it("have expected urls", () => {
+        expect(all.map((x) => x.url())).toMatchSnapshot();
+      });
+    });
   }
-  const repos = () => Array.from(view.repos());
-  it("there is one repository", () => {
-    expect(repos()).toHaveLength(1);
-  });
 
-  const repo = () => assertNotNull(repos()[0]);
-  it("repo matches snapshot", () => {
-    expect(repo()).toMatchSnapshot();
-  });
+  function has(name, method) {
+    it(`has ${name}`, () => {
+      const element = method();
+      let snapshot;
+      if (element != null && typeof element.type === "function") {
+        // element is an Entity. Entities have pointers to the RelationalView,
+        // and it would pollute our snapshot horribly. Just show the url.
+        snapshot = {url: ((element: any): R.Entity).url()};
+      } else {
+        snapshot = element;
+      }
+      expect(snapshot).toMatchSnapshot();
+    });
+  }
 
-  it("repo retrievable by address", () => {
-    expect(view.repo(repo().address)).toEqual(repo());
+  describe("RelationalView", () => {
+    function hasEntityMethods<T: R.Entity>(
+      name,
+      getAll: () => Iterator<T>,
+      get: (x: any) => ?T
```

Comment:
If `Entity` is made a superclass, then you can replace this `any` with:

```js
      get: (x: $Call<$PropertyType<T, "address">>) => ?T
```

(You should be able to do it with a union, too, but Flow complains,
which I think is due to that same bug.)

Reply:
Done